libraries/chain/block.cpp

```diff
@@ -111,4 +112,25 @@ namespace eosio { namespace chain {
       return validate_and_extract_block_extensions( block_extensions );
    }
 
+   signed_block_ptr pruned_block::to_signed_block() const {
+      if (prune_state != prune_state_type::complete_legacy)
+         return signed_block_ptr{};
+
+      auto result = std::make_shared<signed_block>(*static_cast<const signed_block_header*>(this));
+      result->block_extensions = this->block_extensions;
+
+      auto visitor = overloaded{
+          [](const transaction_id_type &id) -> transaction_receipt::trx_type { return id; },
+          [](const pruned_transaction &trx) -> transaction_receipt::trx_type {
+             auto& prunable_data = trx.get_prunable_data().prunable_data;
+             EOS_ASSERT(prunable_data.contains<prunable_transaction_data::full_legacy>(), tx_not_full_legacy, "pruned_transaction does not contain the full legacy data");
```

Comment:
There is no need for this assertion or the new `tx_not_full_legacy` exception. Assuming the invariants of the `pruned_block` have not been violated, then this assertion should never fail because you already checked that `prune_state` must be `prune_state_type::complete_legacy` by this point. And even if there was a programming error that caused this invariant to fail to hold, the `get` function on the `static_variant` would throw an exception anyway.